build/build.go

```diff
@@ -615,7 +615,102 @@ func toSolveOpt(ctx context.Context, di DriverInfo, multiDriver bool, opt Option
 	return &so, releaseF, nil
 }
 
+// ContainerConfig is configuration for a container to run.
+type ContainerConfig struct {
+	ResultCtx *ResultContext
+	Args      []string
+	Env       []string
+	User      string
+	Cwd       string
+	Tty       bool
+	Stdin     io.ReadCloser
+	Stdout    io.WriteCloser
+	Stderr    io.WriteCloser
+}
+
+// ResultContext is a build result with the client that built it.
+type ResultContext struct {
+	Client   *client.Client
+	SolveOpt client.SolveOpt
```

Comment:
Do you even need `SolveOpt` in here? The whole build is based on `Res`. If `SolveOpt` contains output options it may be even wrong to reuse it here.

Reply:
Thank you for the comment. fixed this.